terror_error.go

```diff
@@ -291,3 +296,35 @@ func (e *Error) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func (e *Error) WarpCauseError(err error) *Error {
```

Comment:
Prefer to use a shorter method name `Wrap` (and there is a typo in your signature).